filestore/src/main/java/io/syndesis/filestore/impl/SqlFileStore.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.filestore.impl;
+
+import io.syndesis.filestore.FileStore;
+import io.syndesis.filestore.FileStoreException;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.exceptions.CallbackFailedException;
+
+import java.io.InputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Implementation of a {@code FileStore} backed by a SQL database.
+ */
+public class SqlFileStore implements FileStore {
+
+    enum DatabaseKind {
+        PostgreSQL, H2
```

Comment:
Most of our DB testing is done on Derby, can we add support for that too? Not sure we will use H2, we may want to drop that to simplify maintenance.